src/main/webapp/site/src/app/teacher/end-run-dialog/end-run-dialog.component.html

```diff
@@ -0,0 +1,18 @@
+<h2 class="mat-dialog-title"><ng-container>End Run</ng-container>: {{ run.name }} <span class="mat-caption">(<ng-container i18n>Run ID</ng-container>: {{ run.id }})</span></h2>
+<mat-dialog-content class="mat-dialog-content--scroll" fxLayout="column">
+  <p *ngIf="!isError && !isEnded" i18n>Are you sure you want to end this run? Ending the run will prevent your students from working on the project. You can continue to grade their work.</p>
+  <p *ngIf="isEnded" i18n>The run has been successfully ended.</p>
+  <p *ngIf="isError" i18n>There was an error ending this run. Please try again later.</p>
+</mat-dialog-content>
+<mat-dialog-actions fxLayoutAlign="end" fxLayoutGap="8px">
+  <button type="submit"
+          mat-flat-button
+          color="primary"
+          [disabled]="isEnding"
+          (click)="endRun()"
+          *ngIf="!isError && !isEnded">
+    <mat-progress-bar *ngIf="isEnding" mode="indeterminate"></mat-progress-bar>
+    <ng-container i18n>End Run</ng-container>
+  </button>
+  <button mat-button mat-dialog-close [disabled]="isEnding" i18n>Close</button>
```

Comment:
Can we make the close button text read 'Cancel' by default and then switch to 'Close' after the run has been ended?
```suggestion
  <button mat-button mat-dialog-close [disabled]="isEnding" i18n>
      <ng-container *ngIf="!isError && !isEnded" i18n>Cancel</ng-container>
      <ng-container *ngIf="isError || isEnded" i18n>Close</ng-container>
  </button>
```